### DISCUSSION TITLE
Automatic refresh or Manuel refresh button

### DISCUSSION THREAD

**Discussion by davidolrik** [mcndt/obsidian-toggl-integration#42] [Feature Requests]
I use an external application to start and stop timers, and those changes are not reflected in Obsidian before I reload or relaunch.

To solve this, you could add an automatic refresh of the Toggl data, at most once a minute while the Toggl widget is visible in Obsidian.

Or a quick fix could simply be a manuel refresh button.

**Comment by davidolrik** [comment]
This issue of not refreshing regularly also triggers some subtle bugs.

E.g. I had an existing page with a weekly summary, the summary was in sync with Toggl in the morning.
I then added a few time entries out side of Obsidian, and added a monthly summary to the page.

This resulted in the sidebar and the weekly summary being "frozen" with the data from this morning, and the monthly summary being up to date.

Then a relaunch to get sidebar, weekly and monthly to be in sync.

**Comment by mcndt** [comment]
To resync the reports it suffices to open and close the notes I think. Let me know if that is not the case. 

As to your suggestion for reports, I intentionally made the reports unreactive for now because it would have added a lot of complexity (e.g. what are the conditions for refreshing, should reports be cached,what if someone manually adds timers in the past using Toggl.com, etc). I guess your use case is having a report note always open to see how your week updates? On a related note, how does track.toggl.com handle these updates?

About the sidebar, could you elaborate in which cases the project list doesn’t update? Is it when manually adding time entries to your day externally, or also when starting and stopping a live timer externally?

I mainly wanted to push this update out while I’m between semesters so I can use it when I’m on a tight schedule again, but I’ll try my best to fix any major blockers caused by the update.

**Reply by davidolrik** [reply]
I rarely use track.toggl.com, their API is really nice but their apps are not.

I usually have a timer running all the time when working and I've just tested changing from one project to another, and that makes the sidebar update as it should. 🤔 

I also tried changing to a new timer of the same project, and then deleting the previous timer and changing the current timer to match the just deleted timer, and that also updated the sidebar.

I'm not able to reproduce what I saw earlier, maybe a bad network connection left something in a bad state, but that is just guessing.
If I run into the problem again I'll let you know.

I don't believe there are any major blockers, it's a really nice update 👍🏽

**Reply by mcndt** [reply]
Time heals all bugs... (I wish haha)

Don't hesitate to report if you can reproduce a bug! Even if I can't work on it immediately, it's good to have them documented.
